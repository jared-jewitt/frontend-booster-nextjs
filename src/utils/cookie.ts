interface Options {
  path?: string;
  domain?: string;
  secure?: boolean;
  days?: number;
}

export const cookie = {
  get(name: string): string {
    const cookie = document.cookie.match(`(?:(?:^|.*; *)${name} *= *([^;]*).*$)|^.*$`)[1];
    if (cookie) return decodeURIComponent(cookie);
  },
  set(name: string, value: string | number | boolean, options: Options = {}): void {
    if (options.days) {
      options["max-age"] = options.days * 60 * 60 * 24;
      delete options.days;
    }

    document.cookie =
      name + "=" + encodeURIComponent(value) + Object.entries(options).reduce((acc, [k, v]) => `${acc}; ${k}=${v}`, "");
  },
  delete(name: string): void {
    this.set(name, "", { days: -1 });
  },
};
