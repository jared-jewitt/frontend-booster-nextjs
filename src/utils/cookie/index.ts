interface IOptions {
  path?: string;
  domain?: string;
  secure?: boolean;
  days: number;
}

export const Cookie = {
  get(name: string): string {
    const cookie = document.cookie.match(`(?:(?:^|.*; *)${name} *= *([^;]*).*$)|^.*$`)[1];
    if (cookie) return decodeURIComponent(cookie);
  },
  set(name: string, value: string | number | boolean, options: IOptions): void {
    options["max-age"] = options.days * 60 * 60 * 24;
    delete options.days;

    document.cookie =
      name + "=" + encodeURIComponent(value) + Object.entries(options).reduce((acc, [k, v]) => `${acc}; ${k}=${v}`, "");
  },
  delete(name: string, options?: Pick<IOptions, "path" | "domain">): void {
    this.set(name, "", { ...options, days: -1 });
  },
};
