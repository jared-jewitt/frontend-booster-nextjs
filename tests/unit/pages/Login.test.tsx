import React from "react";
import Login from "@/pages/login";
import { Cookie } from "@/utils";
import { render } from "../../utils";

describe("Login page", () => {
  beforeAll(() => {
    Cookie.delete("isAuthenticated", { path: "/" });
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Login />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
