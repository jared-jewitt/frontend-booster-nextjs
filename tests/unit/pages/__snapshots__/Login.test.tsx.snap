// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="root undefined"
    >
      <a
        href="/login"
      >
        Login
      </a>
    </div>
    <main
      class="main root"
    >
      <div
        style="display: inline-block;
  max-width: 100%;
  overflow: hidden;
  position: relative;
  box-sizing: border-box;
  margin: 0;"
      >
        <div
          style="box-sizing: border-box;
  display: block;
  max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjIwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="max-width: 100%;
  display: block;"
          />
        </div>
        <img
          alt="Logo"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  box-sizing: border-box;
  padding: 0;
  margin: auto;
  display: block;
  width: 0;
  height: 0;
  min-width: 100%;
  max-width: 100%;
  min-height: 100%;
  max-height: 100%;"
        />
      </div>
      <br />
      <button>
        Simulate Login
      </button>
    </main>
  </div>
</DocumentFragment>
`;
